coretex/cli/docker.py

```diff
@@ -0,0 +1,92 @@
+import subprocess
+
+
+class DockerException(Exception):
+
+    """
+        Exception which is raised due to any unexpected behaviour with docker module
+    """
+
+    pass
+
+
+def createNetwork(name: str) -> None:
+    try:
+        subprocess.run(["docker", "network", "create", "--driver", "bridge", name], check=True)
+    except:
+        raise DockerException(f"Failed to create network: {name}")
+
+
+def removeNetwork(name: str) -> None:
+    try:
+        subprocess.run(["docker", "network", "rm", name], check = True)
+    except:
+        raise DockerException(f"Failed to remove network: {name}")
+
+
+def runNode(
+        name: str,
+        dockerImage: str,
+        imageType: str,
+        serverUrl: str,
+        storagePath: str,
+        nodeAccessToken: str,
+        nodeRam: str,
+        nodeSwap: str,
+        nodeSharedMemory: str,
+        restartPolicy: str,
+        ports: str,
+        capAdd: str,
+    ) -> bool:
+
+    runCommand = [
+        "docker", "run", "-d",
+        "--env", f"CTX_API_URL={serverUrl}",
+        "--env", f"CTX_STORAGE_PATH={storagePath}",
+        "--env", f"CTX_NODE_ACCESS_TOKEN={nodeAccessToken}",
+        "--restart", restartPolicy,
+        "-p", ports,
+        "--cap-add", capAdd,
+        "--network", name,
+        "--memory", nodeRam,
+        "--memory-swap", nodeSwap,
+        "--shm-size", nodeSharedMemory,
+        "--name", name,
+    ]
+
+    if imageType == "gpu":
+        runCommand.extend(["--gpus", "all"])
+
+    runCommand.append(dockerImage)
+
+    subprocess.run(runCommand, check=True)
```

Comment:
Create a function which takes in command as a parameter (and any additional one), and wrap it in `try/except` block which will raise `DockerException` if something failes alongside a reason message. Call that everywhere instead of directly calling subprocess module.